core/src/main/java/org/ehcache/events/CacheEventNotificationServiceImpl.java

```diff
@@ -59,13 +60,16 @@
   private final StoreListener<K, V> storeListener = new StoreListener<K, V>();
   private final Store<K, V> store;
   private final TimeSource timeSource;
+  private final ConcurrentHashMap<K, ConcurrentLinkedQueue<CacheEventWrapper>> keyBasedEventQueueMap = new ConcurrentHashMap<K, ConcurrentLinkedQueue<CacheEventWrapper>>();
```

Comment:
This map is populated only in case of `ORDERED` event firing and is cleaned up as and when events are fired and the event queues, containing events per key, are emptied. But this is still essentially a 1-1 mapping. Is this acceptable?